src/EditorFeatures/Core.Wpf/InlineRename/UI/Adornment/RenameFlyoutViewModel.cs

```diff
@@ -246,11 +223,26 @@ protected virtual void Dispose(bool disposing)
             }
         }
 
+        private void ComputeRenameFile()
+        {
+            // If replacementText is invalid, we won't rename the file.
+            RenameFileFlag = _isReplacementTextValid && AllowFileRename && _session.Options.RenameFile;
```

Comment:
Where is the check for "thing being renamed is actually a class that represents the file name"? Is that part of the `InlineRenameFileRenameInfo.Allowed` check on line 49?

Reply:
Correct. That portion is calculated based on the session start